django/core/cache/backends/db.py

```diff
@@ -202,15 +202,29 @@ def _base_set(self, mode, key, value, timeout=DEFAULT_TIMEOUT):
                 return True
 
     def delete(self, key, version=None):
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
+        self.delete_many([key], version)
+
+    def delete_many(self, keys, version=None):
+        key_map = {self.make_key(key, version): key for key in keys}
```

Comment:
This should just be a list of the made keys, as the values aren't used and it's then converted to a list as query params

Reply:
#10257 is being worked on in parallel. I think that this is here because it was erroneously copied as-is from the implementation of `get_many()` which needs to map back to the original key. 

Unfortunately it effectively looks as though this PR has effectively been duplicated in that one due to my [comment](https://github.com/django/django/pull/10257#discussion_r207163455) being misunderstood - although it makes sense to avoid duplicating the key mapping and validation twice, so the implementation seems fine. Maybe both of these should just be rolled into the same PR?

Reply:
Single PR would make things easier to understand

Reply:
Thank you very much for your review.
The two PRs are confusing. So I am going to close the PRs and merge theme to new one PR revised according to your advices.

Reply:
@MyungSeKyo - you could have kept #10257 open, updating the title and commits, and only closed this PR.